File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    const bool isIsomorphic(const string s, const string t) {
        map<char, char> sToT;
        map<char, char> tToS;
        for (int i = 0; i < s.size(); ++i) {
            if (sToT.contains(s[i]) && sToT[s[i]] != t[i]) return false;
            if (tToS.contains(t[i]) && tToS[t[i]] != s[i]) return false;
            sToT[s[i]] = t[i];
            tToS[t[i]] = s[i];
        }
        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.go
func isIsomorphic(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }

    sToT := make(map[byte]byte)
    tToS := make(map[byte]byte)

    for i := 0; i < len(s); i++ {
        val, present := sToT[s[i]]
        if present && val != t[i] {
            return false
        }
        sToT[s[i]] = t[i]

        val, present = tToS[t[i]]
        if present && val != s[i] {
            return false
        }
        tToS[t[i]] = s[i]
    }
    return true
}